feat: add note about root_dir for nvim-lspconfig

martenmatrix opened the pull request:
`nvim-lspconfig` is the official collection of LSP server configurations by NeoVim. Therefore, the majority of NeoVim users use those defaults. It should be included how to prevent attaching `ts_ls` and `denols` to a buffer.

Additionally, [`single_file_support` was renamed to `workspace_required` in NeoVim](https://github.com/neovim/neovim/pull/31824).

martenmatrix committed:
feat: add note about root_dir for nvim-lspconfig

martenmatrix committed:
chore: format

martenmatrix committed:
feat: single_file_support was renamed to workspace_required

thisisjofrank committed:
Merge branch 'main' into patch-1

bartlomieju reviewed (changes requested):
Thanks for the contribution! The `single_file_support` → `workspace_required` update is needed, but there are a few issues to address:

### 1. `root_dir` example is incorrect

The added code block has:
```lua
root_dir = {"package.json"},
```
But `root_dir` expects a function, not a table of strings. It should use `root_pattern` from lspconfig:
```lua
root_dir = require("lspconfig").util.root_pattern("package.json"),
```
As written, this would break users' configs.

### 2. Kickstart.nvim section not updated

The kickstart.nvim example further down the page still uses `single_file_support = false` — this should also be updated to `workspace_required = true` for consistency.

### 3. Fragile GitHub line-number link

The link to `lsp/ts_ls.lua#L56-L68` will break as that file changes. Consider linking to the file without line numbers, or just describing the behavior inline.

martenmatrix committed:
fix: address review comments on nvim-lspconfig section

- Use root_pattern() for root_dir instead of a plain table
- Update Kickstart.nvim block to use workspace_required = true instead of single_file_support = false
- Remove fragile line-number anchor from lsp/ts_ls.lua link

martenmatrix committed:
Merge branch 'main' into patch-1
